generate.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+const VERSION = 0.1;
+const TYPE_GIT = 'git';
+
+$extensions = [];
+
+// --------------------------------------------------------------- //
+// Parse the repositories.json file to extract extension locations //
+// --------------------------------------------------------------- //
+$repositories = json_decode(file_get_contents('repositories.json'), true);
+if (JSON_ERROR_NONE !== json_last_error()) {
+    echo 'The repositories.json file is not a valid JSON file.', PHP_EOL;
+    die;
+}
+
+$gitRepositories = [];
+foreach ($repositories as $repository) {
+    $type = $repository['type'] ?? null;
+    $url = $repository['url'] ?? null;
+    if (null === $type || null === $url) {
+        continue;
+    }
+    if (TYPE_GIT === $type) {
+        $gitRepositories[] = $url;
+    }
+}
+
+// ---------------------------------------- //
+// Clone git repository to extract metadata //
+// ---------------------------------------- //
+foreach ($gitRepositories as $key => $gitRepository) {
+    echo 'Processing ', $gitRepository, ' repository', PHP_EOL;
+    exec("git clone --quiet --single-branch --no-tags {$gitRepository} /tmp/extensions/{$key}");
```

Comment:
```suggestion
    exec("git clone --quiet --depth 1 --no-tags {$gitRepository} /tmp/extensions/{$key}");
```

No need to clone the whole history for that

Reply:
I've added the `depth` option and kept the `single-branch` also.